src/coreclr/jit/runtimelookup.cpp

```diff
@@ -322,8 +321,8 @@ PhaseStatus Compiler::fgExpandRuntimeLookups()
                 fgRemoveRefPred(block, prevBb);
                 fgAddRefPred(block, fastPathBb);
                 fgAddRefPred(block, fallbackBb);
-                nullcheckBb->bbJumpDest = fallbackBb;
-                fastPathBb->bbJumpDest  = block;
+                nullcheckBb->bbJumpDest = fastPathBb;
+                fallbackBb->bbJumpDest  = block;
```

Comment:
What if you mark `fallBackBb->bbSetRunRarely()`. I had to do similar thing to re-arrange in https://github.com/dotnet/runtime/pull/80297.

Reply:
```asm
G_M10050_IG01:
       sub      rsp, 40
                                                ;; size=4 bbWeight=1 PerfScore 0.25
G_M10050_IG02:
       mov      rcx, qword ptr GS:[0x0058]
       mov      rcx, qword ptr [rcx+30H]
       mov      edx, dword ptr [rcx+48H]
       cmp      edx, 2
       jl       SHORT G_M10050_IG04
                                                ;; size=21 bbWeight=1 PerfScore 6.25
G_M10050_IG03:
       mov      rcx, qword ptr [rcx+50H]
       cmp      qword ptr [rcx+10H], 0
       jne      SHORT G_M10050_IG05
                                                ;; size=11 bbWeight=0.90 PerfScore 5.42
G_M10050_IG04:
       mov      rcx, 0xD1FFAB1E
       mov      edx, 29
       mov      r8d, 2
       call     CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED
       mov      qword ptr [rsp+20H], rax
                                                ;; size=31 bbWeight=0.09 PerfScore 0.25
G_M10050_IG05:
       mov      rax, qword ptr [rsp+20H]
       mov      eax, dword ptr [rax+8CH]
                                                ;; size=11 bbWeight=1 PerfScore 3.00
G_M10050_IG06:
       add      rsp, 40
       ret
                                                ;; size=5 bbWeight=1 PerfScore 1.25
```

vs.

```asm
G_M10050_IG01:
       sub      rsp, 40
                                                ;; size=4 bbWeight=1 PerfScore 0.25
G_M10050_IG02:
       mov      rax, qword ptr GS:[0x0058]
       mov      rax, qword ptr [rax+30H]
       mov      ecx, dword ptr [rax+48H]
       cmp      ecx, 2
       jl       SHORT G_M10050_IG06
                                                ;; size=21 bbWeight=1 PerfScore 6.25
G_M10050_IG03:
       mov      rax, qword ptr [rax+50H]
       cmp      qword ptr [rax+10H], 0
       je       SHORT G_M10050_IG06
                                                ;; size=11 bbWeight=0.90 PerfScore 5.42
G_M10050_IG04:
       mov      r9, qword ptr [rsp+20H]
       mov      eax, dword ptr [r9+8CH]
                                                ;; size=12 bbWeight=1 PerfScore 3.00
G_M10050_IG05:
       add      rsp, 40
       ret
                                                ;; size=5 bbWeight=1 PerfScore 1.25
G_M10050_IG06:
       mov      rcx, 0xD1FFAB1E
       mov      edx, 29
       mov      r8d, 2
       call     CORINFO_HELP_GETSHARED_NONGCTHREADSTATIC_BASE_NOCTOR_OPTIMIZED
       mov      r9, rax
       mov      qword ptr [rsp+20H], r9
       jmp      SHORT G_M10050_IG04
                                                ;; size=36 bbWeight=0 PerfScore 0.00
```